tutorials/plot_receptive_field.py

```diff
@@ -0,0 +1,228 @@
+
+fig, ax = plt.subplots()
+ax.pcolormesh(delays_sec, freqs, weights, cmap='RdBu_r')
+ax.set(title='Simulated STRF', xlabel='Time Lags (s)', ylabel='Frequency (Hz)')
+plt.setp(ax.get_xticklabels(), rotation=45)
+plt.autoscale(tight=True)
+mne.viz.tight_layout()
+
+
+###############################################################################
+# Simulate a neural response
+# --------------------------
+#
+# Using this receptive field, we'll create an artificial neural response to
+# a stimulus.
+
+# Reshape audio to split into epochs, then make epochs the first dimension.
+n_epochs, len_epoch = 30, 3
+audio = audio[:, :int(len_epoch * sfreq * n_epochs)]
+X = audio.reshape([n_freqs, n_epochs, -1]).swapaxes(0, 1)
+n_times = X.shape[-1]
+
+# Delay the spectrogram according to delays so it can be combined w/ the STRF
+# Lags will now be in axis 1, then we reshape to vectorize
+X_del = delay_time_series(X, tmin, tmax, sfreq, newaxis=1)
+X_del = X_del.reshape([n_epochs, -1, n_times])
+n_features = X_del.shape[1]
+weights_sim = weights.ravel()
+
+# Simulate a neural response to the sound, given this STRF
+y = np.zeros([n_epochs, n_times])
+for ii, iep in enumerate(X_del):
+    # Simulate this epoch and add random noise
+    noise_amp = .0005
+    y[ii] = np.dot(weights_sim, iep) + noise_amp * rng.randn(n_times)
+
+# Plot the first 2 trials of audio and the simulated electrode activity
+X_plt = scale(np.hstack(X[:2]).T).T
+y_plt = scale(np.hstack(y[:2]))
+time = np.arange(X_plt.shape[-1]) / sfreq
+fig, axs = plt.subplots(2, 1, figsize=(10, 10), sharex=True)
+axs[0].pcolormesh(time, freqs, X_plt, vmin=0, vmax=4, cmap='viridis')
+axs[0].set_title('Input auditory features')
+axs[0].set(ylim=[freqs.min(), freqs.max()], ylabel='Frequency (Hz)')
+axs[1].plot(time, y_plt)
+axs[1].set(xlim=[time.min(), time.max()], title='Simulated response',
+           xlabel='Time (s)', ylabel='Activity (a.u.)')
+mne.viz.tight_layout()
+
+
+###############################################################################
+# Fit a model to recover this receptive field
+# -------------------------------------------
+#
+# Finally, we'll use the :class:`mne.decoding.ReceptiveField`class to recover
+# the linear receptive field of this signal. Note that properties of the
+# receptive field (e.g. smoothness) will depend on the autocorrelation in the
+# inputs and outputs.
+
+# Create training and testing data
+train, test = range(20), 21
+X_train = X[train]
+X_test = X[test]
+y_train = y[train]
+y_test = y[test]
+
+# Model the simulated data as a function of the spectrogram input
+alphas = np.logspace(-4, 0, 10)
+scores = np.zeros(len(alphas))
+models = []
+for ii, alpha in enumerate(alphas):
+    mod = ReceptiveField(tmin, tmax, sfreq, freqs, model=Ridge(alpha))
+    mod.fit(X_train, y_train)
+
+    # Now make predictions about the model output, given input stimuli.
+    y_pred = mod.predict(X_test).squeeze()
+    scores[ii] = r2_score(y_pred, y_test[~mod.mask_remove])
```

Comment:
I had a conversation w/ Frederic the other day about this. I think there are pros and cons to both. Maybe we can discuss at the sprint about whether we want to standardize this for all examples in the future?